File: app/model/Tasks.js
Ext.define('todo.model.Tasks', { 
    extend: 'Ext.data.Model', 
    
    idProperty: 'id',

    fields: [ {name: 'id', type: 'int' },
    		      {name: 'project_id', type:'int'},
     		      {name: 'user_id', type:'int'},
              {name: 'name',  type: 'string'},
              {name: 'description', type:'string'},
              {name: 'created', type:'string'},
              {name: 'updated', type:'string'},
              {name: 'labor', type: 'float'},
              {name: 'status', type: 'string'},
              {name: 'priority', type: 'string'},
              {name: 'username', type:'int'} ],

    validators: {
        name: { type:'length', min: 4 }
    } 
});

File: api-go/actions/tasks.go
package actions

import (
	"../app"
)

type TaskModel struct {
	Id int64 `json "id"`
	ProjectId int64 `json "project_id"`
	UserId int64 `json "user_id"`
	Name string `json "name"`
	Description string `json "description"`
	Created string `json "created"`
	Updated string `json "updated"`
	Labor float64 `json "labor"`
	Status string `json "status"`
	Priority string `json "priority"`
	Username string `json "username"` 
}

type TaskRequestModel struct {
	Id int64 `json "id"`
	Year int64 `json "year"`
	Month int64 `json "month"`
	User int64 `json "user"`
}

type Task struct {
	app.Action
}


func (t Task) Get() {

}

func (t Task) Post() {

}

func (t Task) Put() {

}

func (t Task) Delete() {

}
